front/src/reducers/osrdconf2/common/testUtils.ts

```diff
@@ -0,0 +1,674 @@
+  });
+
+  it('should handle updateDestinationDate', () => {
+    const store = createStore(slice);
+    const newDestinationDate = '10/10/2023';
+    store.dispatch(slice.actions.updateDestinationDate(newDestinationDate));
+    const state = store.getState()[slice.name];
+    expect(state.destinationDate).toBe(newDestinationDate);
+  });
+
+  it('should handle updateDestinationTime', () => {
+    const store = createStore(slice);
+    const newDestinationTime = '10:10:30';
+    store.dispatch(slice.actions.updateDestinationTime(newDestinationTime));
+    const state = store.getState()[slice.name];
+    expect(state.destinationTime).toBe(newDestinationTime);
+  });
+
+  it('should handle updateItinerary', () => {
+    const store = createStore(slice);
+    const newItinerary = testDataBuilder.buildGeoJson();
+    store.dispatch(slice.actions.updateItinerary(newItinerary));
+    const state = store.getState()[slice.name];
+    expect(state.geojson).toBe(newItinerary);
+  });
+
+  it('should handle updateFeatureInfoClickOSRD', () => {
+    const store = createStore(slice);
+    const newFeatureClick = testDataBuilder.buildFeatureInfoClick();
+    store.dispatch(slice.actions.updateFeatureInfoClickOSRD(newFeatureClick));
+    const state = store.getState()[slice.name];
+    expect(state.featureInfoClick).toStrictEqual({
+      ...newFeatureClick,
+      feature: omit(newFeatureClick.feature, ['_vectorTileFeature']),
+    });
+  });
+
+  it('should handle updateGridMarginBefore', () => {
+    const store = createStore(slice);
+    const newGridMarginBefore = 5;
+    store.dispatch(slice.actions.updateGridMarginBefore(newGridMarginBefore));
+    const state = store.getState()[slice.name];
+    expect(state.gridMarginBefore).toStrictEqual(newGridMarginBefore);
+  });
+
+  it('should handle updateGridMarginAfter', () => {
+    const store = createStore(slice);
+    const newGridMarginAfter = 5;
+    store.dispatch(slice.actions.updateGridMarginAfter(newGridMarginAfter));
+    const state = store.getState()[slice.name];
+    expect(state.gridMarginAfter).toStrictEqual(newGridMarginAfter);
+  });
+
+  it('should handle updatePowerRestrictionRanges', () => {
+    const store = createStore(slice);
+    const newPowerRestrictionRanges = testDataBuilder.buildPowerRestrictionRanges();
+    store.dispatch(slice.actions.updatePowerRestrictionRanges(newPowerRestrictionRanges));
+    const state = store.getState()[slice.name];
+    expect(state.powerRestrictionRanges).toStrictEqual(newPowerRestrictionRanges);
+  });
+
+  it('should handle updateTrainScheduleIDsToModify', () => {
+    const store = createStore(slice);
+    const newTrainScheduleIDsToModify = [10, 2];
+    store.dispatch(slice.actions.updateTrainScheduleIDsToModify(newTrainScheduleIDsToModify));
+    const state = store.getState()[slice.name];
+    expect(state.trainScheduleIDsToModify).toStrictEqual(newTrainScheduleIDsToModify);
+  });
+};
+
+function commonConfTestDataBuilder() {
```

Comment:
Don't you want to move this function to the top of the file ? Thus you can remove the eslint warning